provider/scripts/offline_store_spark_runner.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+from datetime import datetime
+
+from pyspark.sql import SparkSession
+
+# Expected format
+
+# job_type: "Transformation", "Materialization", or "Training Set",
+# output_uri: s3://featureform/{type}/{name}/{variant}
+# sql_query: eg. SELECT * FROM source_1 INNER JOIN source_2 ON source_1.id = source_2.id
+# source_list: [s3://featureform/{type}/{name}/{variant}, s3://featureform/{type}/{name}/{variant}...] (implicitly mapped to source_1, source_2, etc)
+
+def execute_sql_query(job_type, output_uri, sql_query, source_list):
+    with SparkSession.builder.appName("Execute SQL Query").getOrCreate() as spark:
+
+        if job_type == "Transformation":
+            for i, source in enumerate(source_list):
+                source_df = spark.read.option("header","true").option("recursiveFileLookup","true").parquet(source)
+                source_df.createOrReplaceTempView(f'source_{i}')
+
+        output_dataframe = spark.sql(sql_query)
+
+        dt = datetime.now()
+        output_uri_with_timestamp = f'{output_uri}/{dt}'
+
+        output_dataframe.write.option("header", "true").mode("overwrite").parquet(output_uri_with_timestamp)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
Seems like this argument is expecting `Transformation`, `Materialization`, or `Training Set`. I would recommend a choices parameter in the function. 
```
parser.add_argument("--job_type", choices=["Transformation", "Materialization", "Training Set"], help="type of job being run on spark") 
```